hpx/runtime/threads/detail/scheduling_loop.hpp

```diff
@@ -420,6 +422,23 @@ namespace hpx { namespace threads { namespace detail
                             num_thread);
                         scheduler.SchedulingPolicy::do_some_work(num_thread);
                     }
+                    else if (HPX_UNLIKELY(state_val == pending_boost))
+                    {
+                        if (HPX_LIKELY(next_thrd == nullptr)) {
+                            // schedule other work
+                            scheduler.SchedulingPolicy::wait_or_add_new(
+                                num_thread, this_state.load() < state_stopping,
+                                idle_loop_count);
+                        }
+
+                        thrd->set_state(pending);
+
+                        // schedule this thread again immediately with boosted
+                        // priority
+                        scheduler.SchedulingPolicy::schedule_thread(thrd,
+                            num_thread, thread_priority_boost);
```

Comment:
[08:07]	hkaiser: heller: also: how about if we change #2487 in the following way
[08:08]	hkaiser: if the busy_loop count would cause background work, we put the thread into next_thrd and reschedule it right away, otherwise we leave it as it is proposed in the PR
[08:09]	heller: sounds good